src/bfs.c

```diff
@@ -0,0 +1,49 @@
+#include "bfs.h"
+
+// Function to initialise the attributes of each node
+struct attributes* init_attributes(struct attributes *nodes, int n) {
+	int i;
+	for (i = 1; i <= n; i++) {
+		nodes[i].parent = NIL;
+		nodes[i].distance = INF;
+		nodes[i].current_color = WHITE;
+	}
+}
+
+// Function to print the value of attributes
+void print_attributes(struct attributes *nodes, int n) {
```

Comment:
Yeah.
Even I dont appreciate naming it `nodes`.
Also, I dont think that `node_attributes` is a good name since its the `nodes[i]` that represents a node's attributes.
If you have any other suggestions, we can work on them else I will add a `TODO` for having a more relevant name than `nodes`.